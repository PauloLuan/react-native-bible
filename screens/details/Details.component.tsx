import * as React from 'react'
import * as S from './details.styles'

export interface DetailsProps {
  testId?: string
}

export const Details = ({
  testId = 'component-details-id',
  ...props
}: DetailsProps) => {
  return (
    <>
      <S.Wrapper testID={testId} {...props}>
        Body...
      </S.Wrapper>
    </>
  )
}
