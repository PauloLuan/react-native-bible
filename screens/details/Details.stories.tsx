import React from 'react'
import { action } from '@storybook/addon-actions'
import { storiesOf } from '@storybook/react-native'
import { Details, DetailsProps } from '.'

const args = {}

storiesOf('Components/Details', module)
  //.addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('default: ', () => <Details {...args} />)
  .add('with some modification: ', () => <Details {...args} />)
