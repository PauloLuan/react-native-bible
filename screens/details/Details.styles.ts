import styled, { css } from 'styled-components/native'

// import { DetailsProps } from '.'

export const Wrapper = styled.View`
  ${() => css`
    flex: 1;
    background-color: blue;
    width: 85px;
    height: 60px;
  `}
`
