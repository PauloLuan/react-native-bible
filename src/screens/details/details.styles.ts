import { RFValue } from 'react-native-responsive-fontsize'
import styled, { css } from 'styled-components/native'

// import { HomeProps } from '.'

export const Wrapper = styled.ScrollView`
  ${() => css`
    flex: 1;
    background-color: ${({ theme }) => theme.colors.background};
    padding: ${RFValue(12)}px;
    width: 100%;
    height: 100%;
  `}
`

export const Headline = styled.Text`
  font-family: ${({ theme }) => theme.fonts.primaryBold};
  font-size: ${RFValue(50)}px;
  font-weight: bold;
`

export const SubHeadline = styled.Text`
  font-family: ${({ theme }) => theme.fonts.primaryLight};
  color: black;
  font-size: ${RFValue(22)}px;
  font-style: italic;
  margin-bottom: ${RFValue(26)}px;
`

export const Verse = styled.Text`
  /* font-family: ${({ theme }) => theme.fonts.primaryMedium}; */
  color: black;
  font-size: ${RFValue(18)}px;
  font-weight: 300;
  line-height: 24px;
  margin: 4px 0;
`

export const VerseNumber = styled.Text`
  font-family: ${({ theme }) => theme.fonts.primaryLight};
  color: black;
  font-size: ${RFValue(12)}px;
  font-weight: 100;
  font-style: italic;
  margin: 4px 0;
`
