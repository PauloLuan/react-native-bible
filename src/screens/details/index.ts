export * from './details.component'
export * as DetailsStyles from './details.styles'
