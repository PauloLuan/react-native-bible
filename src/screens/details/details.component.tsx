import * as React from 'react'
import * as S from './details.styles'
import { Text } from 'react-native'

export interface DetailsProps {
  testId?: string
}

const Item = () => {
  return (
    <>
      <S.Text>Item Sample</S.Text>
    </>
  )
}

export const Details = ({
  testId = 'component-details-id',
  ...props
}: DetailsProps) => {
  const data = [
    'No princípio Deus criou os céus e a terra.',
    'Era a terra sem forma e vazia; trevas cobriam a face do abismo, e o Espírito de Deus se movia sobre a face das águas.',
    'Disse Deus: "Haja luz", e houve luz.',
    'Deus viu que a luz era boa, e separou a luz das trevas.',
    'Deus chamou à luz dia, e às trevas chamou noite. Passaram-se a tarde e a manhã; esse foi o primeiro dia.',
    'Depois disse Deus: "Haja entre as águas um firmamento que separe águas de águas".'
  ]

  return <Text>asdfa sadfadfadf</Text>
}
