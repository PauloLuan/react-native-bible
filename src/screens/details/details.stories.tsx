import { storiesOf } from '@storybook/react-native'
import React from 'react'
import { Details } from './details.component'
// import { ThemeProvider } from 'styled-components/native'
// import theme from '@global/styles/theme'

const args = {
  headline: 'Genesis',
  chapter: 'Capítulo 1',
  verses: [
    'No princípio Deus criou os céus e a terra.',
    'Era a terra sem forma e vazia; trevas cobriam a face do abismo, e o Espírito de Deus se movia sobre a face das águas.',
    'Disse Deus: "Haja luz", e houve luz.',
    'Deus viu que a luz era boa, e separou a luz das trevas.',
    'Deus chamou à luz dia, e às trevas chamou noite. Passaram-se a tarde e a manhã; esse foi o primeiro dia.',
    'No princípio Deus criou os céus e a terra.',
    'Era a terra sem forma e vazia; trevas cobriam a face do abismo, e o Espírito de Deus se movia sobre a face das águas.',
    'Disse Deus: "Haja luz", e houve luz.',
    'Deus viu que a luz era boa, e separou a luz das trevas.',
    'Deus chamou à luz dia, e às trevas chamou noite. Passaram-se a tarde e a manhã; esse foi o primeiro dia.',
    'No princípio Deus criou os céus e a terra.',
    'Era a terra sem forma e vazia; trevas cobriam a face do abismo, e o Espírito de Deus se movia sobre a face das águas.',
    'Disse Deus: "Haja luz", e houve luz.',
    'Deus viu que a luz era boa, e separou a luz das trevas.',
    'Deus chamou à luz dia, e às trevas chamou noite. Passaram-se a tarde e a manhã; esse foi o primeiro dia.',
    'No princípio Deus criou os céus e a terra.',
    'Era a terra sem forma e vazia; trevas cobriam a face do abismo, e o Espírito de Deus se movia sobre a face das águas.',
    'Disse Deus: "Haja luz", e houve luz.',
    'Deus viu que a luz era boa, e separou a luz das trevas.',
    'Deus chamou à luz dia, e às trevas chamou noite. Passaram-se a tarde e a manhã; esse foi o primeiro dia.',
    'No princípio Deus criou os céus e a terra.',
    'Era a terra sem forma e vazia; trevas cobriam a face do abismo, e o Espírito de Deus se movia sobre a face das águas.',
    'Disse Deus: "Haja luz", e houve luz.',
    'Deus viu que a luz era boa, e separou a luz das trevas.',
    'Deus chamou à luz dia, e às trevas chamou noite. Passaram-se a tarde e a manhã; esse foi o primeiro dia.',
    'Depois disse Deus: "Haja entre as águas um firmamento que separe águas de águas".'
  ]
}

storiesOf('Components/Details', module)
  //.addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('default: ', () => (
    // <ThemeProvider theme={theme}>
    <Details {...args} />
    // </ThemeProvider>
  ))
