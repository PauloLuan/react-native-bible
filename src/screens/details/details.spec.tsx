import { waitFor } from '@testing-library/react-native'
import * as React from 'react'
import { renderWithTheme } from '../../utils/test-helper'
import { Details } from './details.component'
// import theme from '../../styles/Theme'

describe('<Details />', () => {
  const testId = 'component-details-id'

  it('should render the Details component', async () => {
    const { queryByTestId, toJSON } = renderWithTheme(
      <Details
        testId={testId}
        headline="teste"
        chapter="1"
        verses={['1', '2']}
      />
    )
    await waitFor(() => expect(queryByTestId(testId)).toBeTruthy())
    expect(toJSON()).toMatchSnapshot()
  })
})
