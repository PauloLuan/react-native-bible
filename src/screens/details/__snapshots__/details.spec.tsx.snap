// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> should render the Details component 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f8f9fa",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100%",
        "paddingBottom": 24,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 24,
        "width": "100%",
      },
    ]
  }
  testID="component-details-id"
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Arial",
          "fontSize": 98,
          "fontWeight": "bold",
        },
      ]
    }
  >
    teste
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 43,
          "fontStyle": "italic",
          "fontWeight": "200",
          "marginBottom": 51,
        },
      ]
    }
  >
    1
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 31,
          "fontWeight": "300",
          "marginBottom": 4,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 4,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 24,
            "fontStyle": "italic",
            "fontWeight": "100",
            "marginBottom": 4,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 4,
          },
        ]
      }
    >
      1
      -
    </Text>
     
    1
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 31,
          "fontWeight": "300",
          "marginBottom": 4,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 4,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 24,
            "fontStyle": "italic",
            "fontWeight": "100",
            "marginBottom": 4,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 4,
          },
        ]
      }
    >
      2
      -
    </Text>
     
    2
  </Text>
</View>
`;
